### PR TITLE
Video upload fix

### LINKED ISSUE #699: Video upload from the asset upload picker is broken on Master
It looks like its due to a PrepareFilesToUpload not accounting for video assets as a type in the uploader method - all assets are assumed to be images. 

I will implement this fix after #698 is merged. Currently the Picker and refactored Instant Upload code paths in PrepareFilesToUpload are completely different - I would like to merge them, and fixing this would be a part of that. It would also be a good first step toward video instant upload.

Edit: Videos will "upload", but they're run through the image path and the uploaded video file is broken.

### PR DESCRIPTION
**Pull request by Jon-Schneider**
This Pull Request:
-  Merges the Gallery Upload and Instant Upload code paths
- Fixes Gallery Video Upload (Issue Fixes #699)

E̶d̶i̶t̶:̶
-̶ ̶A̶d̶d̶e̶d̶ ̶W̶i̶f̶i̶-̶O̶n̶l̶y̶ ̶I̶n̶s̶t̶a̶n̶t̶ ̶U̶p̶l̶o̶a̶d̶,̶ ̶6̶/̶1̶2̶/̶1̶6̶ ̶

E̶d̶i̶t̶ ̶2̶:̶
 ̶ T̶h̶e̶ ̶o̶n̶e̶ ̶t̶h̶i̶n̶g̶ ̶t̶h̶i̶s̶ ̶d̶o̶e̶s̶ ̶n̶o̶t̶ ̶d̶o̶ ̶i̶s̶ ̶s̶t̶o̶p̶ ̶u̶p̶l̶o̶a̶d̶i̶n̶g̶ ̶i̶f̶ ̶y̶o̶u̶ ̶a̶r̶e̶ ̶i̶n̶s̶t̶a̶n̶t̶l̶y̶ ̶u̶p̶l̶o̶a̶d̶i̶n̶g̶ ̶o̶n̶ ̶W̶i̶f̶i̶ ̶a̶n̶d̶ ̶t̶h̶e̶n̶ ̶d̶i̶s̶a̶b̶l̶e̶ ̶w̶i̶f̶i̶ ̶o̶r̶ ̶l̶e̶a̶v̶e̶ ̶t̶h̶e̶ ̶w̶i̶f̶i̶ ̶n̶e̶t̶w̶o̶r̶k̶.̶ ̶I̶'̶m̶ ̶n̶o̶t̶ ̶f̶a̶m̶i̶l̶i̶a̶r̶ ̶e̶n̶o̶u̶g̶h̶ ̶w̶i̶t̶h̶ ̶t̶h̶e̶ ̶f̶i̶l̶e̶ ̶u̶p̶l̶o̶a̶d̶i̶n̶g̶ ̶i̶n̶t̶e̶r̶f̶a̶c̶e̶,̶ ̶b̶u̶t̶ ̶w̶h̶a̶t̶ ̶w̶o̶u̶l̶d̶ ̶b̶a̶s̶i̶c̶a̶l̶l̶y̶ ̶h̶a̶v̶e̶ ̶t̶o̶ ̶h̶a̶p̶p̶e̶n̶ ̶i̶s̶ ̶u̶p̶l̶o̶a̶d̶i̶n̶g̶ ̶f̶i̶l̶e̶s̶ ̶w̶o̶u̶l̶d̶ ̶n̶e̶e̶d̶ ̶t̶o̶ ̶b̶e̶ ̶m̶a̶r̶k̶e̶d̶ ̶a̶s̶ ̶i̶n̶s̶t̶a̶n̶t̶ ̶u̶p̶l̶o̶a̶d̶s̶,̶ ̶t̶h̶e̶n̶ ̶a̶u̶t̶o̶m̶a̶t̶i̶c̶a̶l̶l̶y̶ ̶d̶e̶q̶u̶e̶u̶e̶d̶ ̶a̶n̶d̶ ̶t̶h̶e̶ ̶i̶n̶s̶t̶a̶n̶t̶ ̶u̶p̶l̶o̶a̶d̶ ̶d̶a̶t̶e̶ ̶r̶e̶s̶e̶t̶ ̶t̶o̶ ̶j̶u̶s̶t̶ ̶b̶e̶f̶o̶r̶e̶ ̶t̶h̶e̶ ̶e̶a̶r̶l̶i̶e̶s̶t̶ ̶a̶s̶s̶e̶t̶ ̶b̶e̶i̶n̶g̶ ̶u̶p̶l̶o̶a̶d̶e̶d̶ ̶b̶y̶ ̶t̶h̶e̶ ̶I̶n̶s̶t̶a̶n̶t̶ ̶U̶p̶l̶o̶a̶d̶ ̶m̶a̶n̶a̶g̶e̶r̶ ̶i̶f̶ ̶t̶h̶e̶ ̶r̶e̶a̶c̶h̶a̶b̶i̶l̶i̶t̶y̶ ̶s̶t̶a̶t̶u̶s̶ ̶c̶h̶a̶n̶g̶e̶s̶ ̶t̶o̶ ̶n̶o̶ ̶w̶i̶f̶i̶.̶ ̶ ̶ ̶I̶f̶ ̶w̶e̶ ̶w̶a̶n̶t̶ ̶t̶o̶ ̶b̶r̶e̶a̶k̶ ̶t̶h̶e̶ ̶v̶i̶d̶e̶o̶ ̶u̶p̶l̶o̶a̶d̶ ̶f̶i̶x̶ ̶f̶o̶r̶ ̶t̶h̶e̶ ̶g̶a̶l̶l̶e̶r̶y̶ ̶u̶p̶l̶o̶a̶d̶ ̶o̶u̶t̶ ̶a̶n̶d̶ ̶j̶u̶s̶t̶ ̶i̶n̶c̶l̶u̶d̶e̶ ̶t̶h̶a̶t̶ ̶a̶n̶d̶ ̶w̶o̶r̶r̶y̶ ̶a̶b̶o̶u̶t̶ ̶w̶i̶f̶i̶ ̶i̶n̶s̶t̶a̶n̶t̶ ̶u̶p̶l̶o̶a̶d̶ ̶i̶n̶ ̶a̶ ̶r̶e̶l̶e̶a̶s̶e̶ ̶a̶f̶t̶e̶r̶ ̶3̶.̶5̶ ̶I̶ ̶c̶a̶n̶ ̶m̶o̶d̶i̶f̶y̶ ̶t̶h̶e̶ ̶b̶r̶a̶n̶c̶h̶ ̶a̶c̶c̶o̶r̶d̶i̶n̶g̶l̶y̶.

Edit 3: Removes basic Wifi-only upload.

### CODE REVIEW

**Comment by nasli** [comment]
Yes @Jon-Schneider, is better if you separate the fix from the new wifi feature, and release the wifi feature fully functional allowing instant uploads only with wifi taken into account all different scenarios when you lost your wifi connection

**Comment by Jon-Schneider** [comment]
@nasli Is it possible to dequeue files that have been queued to be uploaded? As far as I know its not possible, but I don't know as well as you.

**Comment by nasli** [comment]
Within next release we are going to use sessions instead of operations(https://github.com/owncloud/ios/issues/662) and we could consider use allowsCellularAccess https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSessionConfiguration_class/index.html#//apple_ref/occ/instp/NSURLSessionConfiguration/allowsCellularAccess.

**Comment by Jon-Schneider** [comment]
Not bad, the only issue with that is if you switch your preference from
wifi to cellular while you have a bunch of wifi-only instant upload tasks
queued it won't be able to change them to upload over mobile data and being
the uploads, but its a very clean solution if we don't mind the sedge case.
I like that the upload tasks handle wifi vs. mobile data themselves because
it means the file upload system doesn't need rewritten.

File: Owncloud iOs Client/Network/Uploads/PrepareFilesToUpload.m [hunk 1]
```diff
@@ -145,117 +138,95 @@ - (void)uploadFileFromGallery:(PHAsset *)assetToUpload andRemoteFolder:(NSString
...
             
-            //We finish all the files of this block
-            DLog(@"self.listOfUploadOfflineToGenerateSQL: %lu", (unsigned long)[self.listOfUploadOfflineToGenerateSQL count]);
+            NSURL *url = info[@"PHImageFileURLKey"];
+            NSString *fileExtension = url.pathExtension;
             
-            //In this point we have all the files to upload in the Array
-            [ManageUploadsDB insertManyUploadsOffline:self.listOfUploadOfflineToGenerateSQL];
+            //Use a temporary name with a date identification
+            NSString *temporaryFileName = [NSString stringWithFormat:@"IMG_%@.%@", [dateFormatter stringFromDate:[NSDate date]], fileExtension];
```
Comment (nasli) [inline Owncloud iOs Client/Network/Uploads/PrepareFilesToUpload.m]: We include here a regression bug with the refactor of instant uploads. Please keep using same name as before. For the new PHAssets a new method was created, you can use it [getComposeNameFromPHAsset](https://github.com/owncloud/ios/blob/master/Owncloud%20iOs%20Client/Utils/FileNameUtils.m#L467)

**Comment by Jon-Schneider** [comment]
Updated to use +[FileNameUtils getComposeNameFromPHAsset] for gallery asset upload filenames.

**Comment by Jon-Schneider** [comment]
Does anyone want to do a review of this pull request? Its pretty straightforward.

**Comment by jesmrec** [comment]
@Jon-Schneider  I will review this PR (from a QA pov) asap.

**Comment by jesmrec** [comment]
Uploaded videos are correctly downloaded and played into the app

Approved

cc @Jon-Schneider @nasli

**Comment by Jon-Schneider** [comment]
Thanks @jesmrec!

Does anyone want to merge this PR so I can move on to Wifi Instant Upload?

**Comment by jesmrec** [comment]
@nasli @mRoRo

**Comment by nasli** [comment]
👍

**Pull request merged**